spring-integration-ip/src/main/java/org/springframework/integration/ip/udp/DatagramPacketMessageMapper.java

```diff
@@ -142,7 +144,20 @@ public DatagramPacket fromMessage(Message<?> message) throws Exception {
 			buffer.put(bytes);
 			bytes = buffer.array();
 		}
-		return new DatagramPacket(bytes, bytes.length);
+		return getDatagramFromHeader(message, bytes, bytes.length);
+	}
+
+	private DatagramPacket getDatagramFromHeader(Message<?> message, byte[] bytes,
+												 int length) throws UnknownHostException {
+		InetAddress inetAddress = InetAddress.getByName(
+				message.getHeaders().get(IpHeaders.IP_ADDRESS, String.class));
```

Comment:
I think it isn't good idea to rely on the `null` from the header.
See the `InetAddress` source code:
```
if (host == null || host.length() == 0) {
        InetAddress[] ret = new InetAddress[1];
        ret[0] = impl.loopbackAddress();
        return ret;
}
```
So, let's move `InetAddress.getByName()` to the next `if`!